File: test/ch21-benchmarks/bench-005-array-sum.go
// BENCH-005: Array Sum (1 million integers) - Go
package main

func arraySum(n int) int {
	sum := 0
	for i := 0; i < n; i++ {
		sum += i
	}
	return sum
}

func main() {
	result := arraySum(1000000)
	_ = result // Expected: 499999500000
}

File: test/ch21-benchmarks/bench-005-array-sum.c
// BENCH-005: Array Sum (1 million integers) - C
#include <stdio.h>

long long array_sum(int n) {
    long long sum = 0;
    for (int i = 0; i < n; i++) {
        sum += i;
    }
    return sum;
}

int main() {
    long long result = array_sum(1000000);
    // Expected: 499999500000
    return 0;
}
